x/interchainstaking/keeper/withdrawal_record.go

```diff
@@ -14,18 +16,40 @@ import (
 	"github.com/quicksilver-zone/quicksilver/x/interchainstaking/types"
 )
 
-func (k *Keeper) GetNextWithdrawalRecordSequence(ctx sdk.Context) (sequence uint64) {
-	store := prefix.NewStore(ctx.KVStore(k.storeKey), nil)
+// InitWithdrawalRecordSequence initializes the sequence.
+func (k *Keeper) InitWithdrawalRecordSequence(ctx sdk.Context) {
+	var accNumber uint64
+	store := ctx.KVStore(k.storeKey)
+
+	bz := k.cdc.MustMarshal(&gogotypes.UInt64Value{Value: 512})
+	store.Set(types.KeyPrefixRequeuedWithdrawalRecordSeq, bz)
+
+	return accNumber
+}
+
+// GetNextWithdrawalRecordSequence returns and increments the global withdrawal record seqeuence.
+func (k *Keeper) GetNextWithdrawalRecordSequence(ctx sdk.Context) uint64 {
+	var sequence uint64
+	store := ctx.KVStore(k.storeKey)
+
 	bz := store.Get(types.KeyPrefixRequeuedWithdrawalRecordSeq)
 	if bz == nil {
-		bz := make([]byte, 8)
-		binary.BigEndian.PutUint64(bz, uint64(2))
-		store.Set(types.KeyPrefixRequeuedWithdrawalRecordSeq, bz)
-		return 1
+		// initialize the account numbers
+		sequence = 0
+	} else {
+		val := gogotypes.UInt64Value{}
+
+		err := k.cdc.Unmarshal(bz, &val)
+		if err != nil {
+			panic(err)
+		}
+
+		sequence = val.GetValue()
 	}
-	sequence = binary.BigEndian.Uint64(bz)
-	binary.BigEndian.PutUint64(bz, sequence+1)
+
+	bz = k.cdc.MustMarshal(&gogotypes.UInt64Value{Value: sequence + 1})
 	store.Set(types.KeyPrefixRequeuedWithdrawalRecordSeq, bz)
+
```

Comment:
@coderabbitai - when I first reviewed, I misread -- this is reading from our chain, and so the code is correct.  We should panic if our store contains corrupted data.

Reply:
@coderabbitai any time.  

We're kinda consciously teaching you as much as we can about the codebase.  @joe-bowman and I are pretty jazzed about your contributions thus far, and everyone on the team has enjoyed working with you.  Brave new world I suppose, welcome, non human colleague/contributor.

Reply:
oh, right I forgot another thing @coderabbitai - in cosmos, we refer to the notion that a chain with problems SHOULD go down instead of continuing to make blocks as "safety over liveness"